################################ General Parameters #############################################
general:
  # Specify the mode of operation for the segmentation pipeline
  # 0: Semantic Segmentation runs in conjunction with Geometric Segmentation
  # 1: Semantic Segmentation runs independently
  # 2: Geometric Segmentation runs independently
  mode_of_operation: 0

  # The path of the high level semantic information of the environment
  # [TODO] - the path looks ugly, possible workaround?
  env_database: "../../../src/visual_sgraphs/config/Environments/LuxVSD/luxvsd_seq02.json"

#################################### Marker Parameters ###########################################
markers:
  # How much the framework should trust the markers (0.1 default, higher for mono setups)
  impact: 0.1

############################## PointCloud Parameters #############################################
pointcloud:
  distance_thresh:
    near: 0.3
    far: 3.0

################################ Common Segmentation Parameters ##################################
seg:
  # Number of points (after filtering) needed to detect geometric objects (e.g., planes)
  # Also, the number of points needed to detect a plane in the pointcloud
  pointclouds_thresh: 400

  # Plane association threshold (ominus)
  plane_association_thresh: 0.2

  # The maximum distance for point to be considered part of a plane
  plane_point_dist_thresh: 0.05

  # The maximum threshold for the dot product of the plane normals to be considered facing each other
  plane_facing_dot_thresh: -0.9

  ransac:
    # The maximum number of planes to be detected in one pointcloud
    max_planes: 1

    # The maximum distance for a point to be considered an inlier of the plane model
    distance_thresh: 0.05

    # The maximum number of iterations for the RANSAC algorithm
    max_iterations: 500

############################ Geometric Segmentation Parameters ####################################
geo_seg:
  # The leaf size for the pointcloud downsampling
  downsample_leaf_size: 0.08

############################## Semantic Segmentation Parameters ####################################
sem_seg:
  # The leaf size for the pointcloud downsampling
  downsample_leaf_size: 0.04

  # The threshold for semantic segmentation probability validation (rejects probability below this)
  prob_thresh: 0.7

  # The maximum height above the main ground to be a valid stepped ground plane (in meters)
  max_step_elevation: 0.5

  # The maximum tilt heuristic for a wall/ground plane (checks respective component of plane equation)
  max_tilt_wall: 0.2
  max_tilt_ground: 0.2

  # The amount of votes (votes are weighted votes) for a plane to be finally classified with a semnatic label
  min_votes: 2.5

  ############################## Room Segmentation Parameters ####################################
room_seg:
  # The method to use for room segmentation
  # 0: Segment geometrically based on closest walls
  # 1: Segment based on free-space clustering (skeleton voxblox)
  # 2: Segment based on GNN
  method: 1

  # Threshold for perpendicularity of wall pairs to be considered a square room (in degrees)
  perpendicularity_thresh: 10.0

  # Parameters for the geometrically-based closest walls room segmentation
  geo_based:
    # The maximum distance from a marker to a wall to be considered part of the room
    marker_wall_distance_thresh: 6.0

  # Parameters for skeleton voxblox free-space room segmentation
  skeleton_based:
    # The minimum number of points needed to form a cluster
    min_cluster_vertices: 5
